targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/i2c_api.c

```diff
@@ -619,51 +644,91 @@ static void i2c_configure_driver_instance(i2c_t *obj)
+        		nrfx_twi_disable(&nordic_nrf5_instance[instance]);
+			nrfx_twi_uninit(&nordic_nrf5_instance[instance]);
+#endif
+
         }
 
         /* Force resource release. This is necessary because mbed drivers don't
          * deinitialize on object destruction.
          */
         NRFX_IRQ_DISABLE((nrfx_get_irq_number((void const*)nordic_nrf5_twi_register[instance])));
 
-        /* Configure driver with new settings. */
-        nrfx_twi_config_t twi_config = {
-            .scl = config->scl,
-            .sda = config->sda,
-            .frequency = config->frequency,
-            .interrupt_priority = APP_IRQ_PRIORITY_LOWEST,
-            .hold_bus_uninit = false
-        };
-
-#if DEVICE_I2C_ASYNCH
-        /* Set callback handler in asynchronous mode. */
-        if (config->handler) {
-
-            /* Initialze driver in non-blocking mode. */
-            nrfx_twi_init(&nordic_nrf5_instance[instance],
-                          &twi_config,
-                          nordic_nrf5_twi_event_handler,
-                          obj);
-        } else {
-
-            /* Initialze driver in blocking mode. */
-            nrfx_twi_init(&nordic_nrf5_instance[instance],
-                          &twi_config,
-                          NULL,
-                          NULL);
+#if DEVICE_I2CSLAVE
+        if(config->is_slave) {
+			/* Configure driver as slave with new settings. */
+        		nrfx_twis_config_t twis_config = {
+        				.addr = { (config->slave_addr >> 1), 0 },
+        				.scl = config->scl,
+					.sda = config->sda,
+					.scl_pull = NRF_GPIO_PIN_NOPULL,
+					.sda_pull = NRF_GPIO_PIN_NOPULL,
+					.interrupt_priority = APP_IRQ_PRIORITY_LOWEST
+        		};
+
+			/* Initialze driver in blocking mode. */
+			nrfx_twis_init(&nordic_nrf5_twis_instance[instance],
+						  &twis_config,
+						  NULL);
+
+			/* Enable peripheral (nrfx_twi_t and nrfx_twis_t are interchangeable). */
+			nrfx_twis_enable(&nordic_nrf5_twis_instance[instance]);
         }
-#else
-        /* Initialze driver in blocking mode. */
-        nrfx_twi_init(&nordic_nrf5_instance[instance],
-                      &twi_config,
-                      NULL,
-                      NULL);
+        else {
 #endif
 
-        /* Enable peripheral. */
-        nrfx_twi_enable(&nordic_nrf5_instance[instance]);
+			/* Configure driver with new settings. */
+			nrfx_twi_config_t twi_config = {
+				.scl = config->scl,
+				.sda = config->sda,
+				.frequency = config->frequency,
+				.interrupt_priority = APP_IRQ_PRIORITY_LOWEST,
+				.hold_bus_uninit = false
+			};
+
+	#if DEVICE_I2C_ASYNCH
+			/* Set callback handler in asynchronous mode. */
+			if (config->handler) {
+
+				/* Initialze driver in non-blocking mode. */
+				nrfx_twi_init(&nordic_nrf5_instance[instance],
+							  &twi_config,
+							  nordic_nrf5_twi_event_handler,
+							  obj);
+			} else {
+
+				/* Initialze driver in blocking mode. */
+				nrfx_twi_init(&nordic_nrf5_instance[instance],
```

Comment:
this is HAL file, isn't it using spaces as defined in astyle config ? I can see this is 8 spaces at least, or even tabs?

As it was , it was following the style.